### PR TITLE
feat(runtime): Add support of `upload_code` for vouchers

### PR DESCRIPTION
**Pull request by breathx**

### CODE REVIEW

**Review by ukint-vs — APPROVED** [review]

File: pallets/gear-voucher/src/lib.rs [hunk 1]
```diff
@@ -417,6 +424,7 @@ pub mod pallet {
             move_ownership: Option<AccountIdOf<T>>,
             balance_top_up: Option<BalanceOf<T>>,
             append_programs: Option<Option<BTreeSet<ProgramId>>>,
+            code_uploading: Option<bool>,
```
Comment (shamilsan) [inline pallets/gear-voucher/src/lib.rs:427]: Please add `code_uploading` arg to docs above.

**Review by shamilsan — APPROVED** [review]

File: pallets/gear-voucher/src/internal.rs [hunk 2]
```diff
@@ -52,7 +57,9 @@ where
                 .ok(),
 
             #[allow(deprecated)]
-            Self::call_deprecated { call: prepaid_call } => {
+            Self::call_deprecated { call: prepaid_call }
+                if !matches!(prepaid_call, PrepaidCall::UploadCode { .. }) =>
```
Comment (ark0f) [inline pallets/gear-voucher/src/internal.rs:61]:
```suggestion
            Self::call_deprecated { call: PrepaidCall::UploadCode { .. } } =>
```

File: pallets/gear-voucher/src/internal.rs [hunk 3]
```diff
@@ -70,6 +77,10 @@ impl<T: Config> Pallet<T> {
         who: &T::AccountId,
         call: &PrepaidCall<BalanceOf<T>>,
     ) -> Option<T::AccountId> {
+        if matches!(call, PrepaidCall::UploadCode { .. }) {
```
Comment (ark0f) [inline pallets/gear-voucher/src/internal.rs:80]: Check is already in `Self::prepaid_call_destination`

File: pallets/gear-voucher/src/lib.rs [hunk 4]
```diff
@@ -435,6 +443,9 @@ pub mod pallet {
             // Flattening move ownership back to current owner.
             let new_owner = move_ownership.filter(|addr| addr.ne(&voucher.owner));
 
+            // Flattening code uploading.
+            let code_uploading = code_uploading.filter(|v| v.ne(&voucher.code_uploading));
```
Comment (ark0f) [inline pallets/gear-voucher/src/lib.rs:447]:
```suggestion
            let code_uploading = code_uploading.filter(|v| v != voucher.code_uploading);
```

**Review by ark0f — APPROVED** [review]

**Pull request merged**